eidas_node/proxy_service/views.py

```diff
@@ -1 +1,257 @@
 """Views of eidas_node.proxy_service."""
+import logging
+from base64 import b64decode, b64encode
+from collections import namedtuple
+from datetime import datetime, timedelta
+from typing import Any, Dict, Optional, Tuple
+from uuid import uuid4
+
+from django.http import HttpRequest, HttpResponse
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+from django.views.generic import TemplateView
+from lxml.etree import XMLSyntaxError
+
+from eidas_node.errors import EidasNodeError, ParseError, SecurityError
+from eidas_node.models import LightRequest, LightResponse, LightToken
+from eidas_node.proxy_service.settings import PROXY_SERVICE_SETTINGS
+from eidas_node.saml import SAMLRequest, SAMLResponse, decrypt_xml
+from eidas_node.storage import LightStorage
+from eidas_node.utils import dump_xml, import_from_module, parse_xml
+
+Error = namedtuple('Error', 'status_code, message')
+
+LOGGER = logging.getLogger('eidas_node.proxy_service')
+
+
+class ProxyServiceRequestView(TemplateView):
+    """
+    Forward service provider's request to an identity provider.
+
+    eIDAS Generic Proxy Service provides the service provider's request as a light request.
+    """
+
+    template_name = 'eidas_node/proxy_service/proxy_service_request.html'
+    error = None  # type: Optional[Error]
+    storage = None  # type: LightStorage
+    light_token = None  # type: LightToken
+    light_request = None  # type: LightRequest
+    saml_request = None  # type: SAMLRequest
+
+    def get(self, *args, **kwargs):
+        """Handle a HTTP GET request."""
+        return self.http_method_not_allowed(*args, **kwargs)
+
+    def post(self, request: HttpRequest) -> HttpResponse:
+        """Handle a HTTP POST request."""
+        try:
+            token_settings = PROXY_SERVICE_SETTINGS.request_token
+            self.light_token = self.get_light_token(
+                token_settings['parameter_name'],
+                token_settings['issuer'],
+                token_settings['hash_algorithm'],
+                token_settings['secret'],
+                token_settings.get('lifetime'))
+            LOGGER.debug('Light Token: %s', self.light_token)
+            self.storage = self.get_light_storage(PROXY_SERVICE_SETTINGS.light_storage['backend'],
+                                                  PROXY_SERVICE_SETTINGS.light_storage['options'])
+            self.light_request = self.get_light_request()
+            LOGGER.debug('Light Request: %s', self.light_request)
+            self.saml_request = self.create_saml_request(PROXY_SERVICE_SETTINGS.identity_provider['request_issuer'])
+            LOGGER.debug('SAML Request: %s', self.saml_request)
+        except EidasNodeError:
+            LOGGER.exception('Bad proxy service request.')
+            self.error = Error(400, _('Bad proxy service request.'))
+
+        response = super().get(request)
```

Comment:
I would rather detect error and return [HttpResponseBadRequest](https://docs.djangoproject.com/en/dev/ref/request-response/#django.http.HttpResponseBadRequest) or `super()`.

Reply:
Done.